Plugins/NativeXr/Source/NativeXr.cpp

```diff
@@ -2209,6 +2214,44 @@ namespace Babylon
                 }
             }
 
+            Napi::Value FillPoses(const Napi::CallbackInfo& info)
+            {
+                const auto spaces = info[0].As<Napi::Array>();
+                auto transforms = info[2].As<Napi::Float32Array>();
+                if (spaces.Length() != transforms.ElementLength() >> 4)
+                {
+                    throw std::runtime_error{"Number of spaces doesn't match number of transforms * 16."};
+                }
+
+                for (uint32_t spaceIdx = 0; spaceIdx < spaces.Length(); spaceIdx++)
+                {
+                    const auto& jointSpace = *spaces[spaceIdx].As<Napi::External<xr::System::Session::Frame::Space>>().Data();
+                    const auto transformMatrix = CreateTransformMatrix(jointSpace, false);
+                    memcpy(transforms.Data() + (spaceIdx << 4), transformMatrix.data(), 4 /*Float32*/ * 16);
```

Comment:
nit: I personally hate constants like this. Can we either express these offsets in terms of `sizeof` results (I think you're trying to do `sizeof(float)`) or at least a named constant that explains what's going on?